### DISCUSSION TITLE
Tap, Pan, Pinch Composed Gesture

### DISCUSSION THREAD

**Discussion by nagamanikandank** [software-mansion/react-native-gesture-handler#3208] [Q&A]
I need to detect, Tap, Pinch & Zoom .
This is my `composed` setting 
`const composed = Gesture.Exclusive(tap, Gesture.Race(pinch,pan));`

When I Tap, only Tap gets detected
When I Pan, only Pan gets detected.
But, when I Pinch, both Pan and Pinch gets detected.

How can I make only Pinch get detected ?

**Comment by Glazzes** [comment]
For my [library](https://github.com/Glazzes/react-native-zoom-toolkit), I do it like this:

```js
const composedTap = Gesture.Exclusive(doubleTap, tap);
const composedGesture = Gesture.Race(pinch, pan, composedTap);
```

**Reply by nagamanikandank** [reply]
Thanks for your response. I have moved away from React Native to focus on other projects. Hence closing this discussion

**Comment by m-bert** [comment]
Hi @nagamanikandank! Could you share a snippet of how do you define those gestures?

**Reply by nagamanikandank** [reply]
I have moved away from React Native to focus on other projects. Hence closing this discussion
